frontend/templates/product/sections/ReviewsSection/index.tsx

```diff
@@ -220,7 +249,30 @@ type RatingCount = {
 };
 
 function formatReviewDate(timeSeconds: number) {
-   return new Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(
-      new Date(timeSeconds * 1000)
-   );
+   const formatter = new Intl.RelativeTimeFormat('en-US', { numeric: 'auto' });
+   const date = new Date(timeSeconds * 1000);
+   let duration = (date.getTime() - new Date().getTime()) / 1000;
+
+   for (let i = 0; i <= DIVISIONS.length; i++) {
+      const division = DIVISIONS[i];
+      if (Math.abs(duration) < division.amount) {
+         return formatter.format(Math.round(duration), division.name);
+      }
+      duration /= division.amount;
+   }
 }
+
+type Division = {
+   amount: number;
+   name: Intl.RelativeTimeFormatUnit;
+};
+
+const DIVISIONS: Division[] = [
+   { amount: 60, name: 'seconds' },
+   { amount: 60, name: 'minutes' },
+   { amount: 24, name: 'hours' },
+   { amount: 7, name: 'days' },
+   { amount: 4.34524, name: 'weeks' },
```

Comment:
Took me a second to figure out that this is the average number of weeks in a month! That was a fun puzzle 😂